### PR TITLE
bugfix/13229-empty-angle-donut

### LINKED ISSUE #13229: Semi circle donut with no data
#### Expected behaviour

When rendering a semi circle donut chart with no data the pie shape should still be a semi circle

#### Actual behaviour

The semi circle becomes a full circle moved downwards

#### Live demo with steps to reproduce

No data: https://jsfiddle.net/y4750g9f/
With data: https://jsfiddle.net/2aj5tsxm/

#### Product version
Latest version

#### Affected browser(s)
All

### PR DESCRIPTION
**Pull request by madepiet**
Fixed #13229, the empty data shape was wrong for pies if a start or end angle or inner size was set.

~~Fixed #13229, drawn path instead of circle in drawEmpty.~~

### CODE REVIEW

File: ts/parts/PieSeries.ts [hunk 1]
```diff
@@ -1051,29 +1051,37 @@ seriesType<Highcharts.PieSeries>(
...
-                this.graph.animate({
+                this.graph.attr({
+                    d: Highcharts.SVGRenderer.prototype.symbols.arc(
+                        centerX,
+                        centerY,
+                        this.center[2] / 2,
+                        0, {
+                            start: start,
+                            end: end,
+                            innerR: options.innerSize
```
Comment (TorsteinHonsi) [inline ts/parts/PieSeries.ts:1078]: This fails when `innerSize` is percent. Instead, use the computed value `this.center[3] / 2`. Test case: 
```js
Highcharts.chart('container', {
    series: [{
        data: [null, null, null],
        startAngle: -90,
        endAngle: 90,
        type: 'pie',
        innerSize: '50%'
    }]
});
```

File: ts/parts/PieSeries.ts [hunk 2]
```diff
@@ -1051,29 +1051,37 @@ seriesType<Highcharts.PieSeries>(
...
                     'stroke-width': options.borderWidth,
-                    cx: centerX,
-                    cy: centerY,
-                    r: this.center[2] / 2,
-                    fill: (options.fillColor as any) || 'none',
-                    stroke: (options.color as any) ||
-                        '${palette.neutralColor20}'
-                }, this.options.animation);
+                    fill: options.fillColor || 'none',
+                    stroke: options.color || '#cccccc'
```
Comment (TorsteinHonsi) [inline ts/parts/PieSeries.ts:1083]: One change that has slipped through into production already, but we should fix it now that we are here. 

1. These presentational properties should not apply in `styledMode`. See https://jsfiddle.net/highcharts/n9s2Ld1a/. 
2. For the default stroke color, use `${palette.neutralColor20}`
3. Add a more specific class name to the graph, suggesting `highcharts-empty-series`.

File: samples/unit-tests/series-pie/pie/demo.js [hunk 3]
```diff
@@ -99,6 +99,29 @@ QUnit.test('Update to negative (#7113) + Empty pie look (#5526)', function (asse
...
+            data: []
+        }
+    })
+
+    var graph = chart.series[0].graph;
+
+    // Issue #13229
+    assert.ok(
+        graph.pathArray,
+        'Path should be drawn instead of a circle.'
```
Comment (pawelfus) [inline samples/unit-tests/series-pie/pie/demo.js:123]: Please add issue number

File: samples/unit-tests/series-pie/pie/demo.js [hunk 4]
```diff
@@ -185,65 +208,4 @@ QUnit.test('Updating point visibility (#8428)', function (assert) {
         isHidden(point.dataLabel),
         'Hidden point should not have a data label'
     );
-});
-
-QUnit.test('Animation - Empty pie', function (assert) {
```
Comment (pawelfus) [inline samples/unit-tests/series-pie/pie/demo.js:190]: Why this test is removed?

File: ts/parts/PieSeries.ts [hunk 5]
```diff
@@ -1051,29 +1051,37 @@ seriesType<Highcharts.PieSeries>(
...
-                        centerY, 0)
+                if (!this.graph) {
+                    this.graph = this.chart.renderer
+                        .arc(centerX, centerY, this.center[1] / 2, 0, start, end)
                         .addClass('highcharts-graph')
                         .add(this.group);
                 }
 
-                this.graph.animate({
+                this.graph.attr({
```
Comment (pawelfus) [inline ts/parts/PieSeries.ts:1070]: Why animation is removed?

File: ts/parts/PieSeries.ts (hunk 2 above)
Comment (pawelfus) [inline ts/parts/PieSeries.ts:1083]: Any reason to change `'${palette.neutralColor20}'` to `#cccccc` ?

File: ts/parts/PieSeries.ts (hunk 5 above)
Comment (madepiet) [reply ts/parts/PieSeries.ts:1070]: Getting rid of the animation from this place resulted from my conversation with @Torstein - the purpose of the series animation is primarily to draw the user's attention and invite him to interact (there is nothing to pay attention to in the empty graphics).

Comment (madepiet) [reply ts/parts/PieSeries.ts:1070]: Getting rid of the animation from this place resulted from my conversation with @TorsteinHonsi - the purpose of the series animation is primarily to draw the user's attention and invite him to interact (there is nothing to pay attention to in the empty graphics).

File: ts/parts/PieSeries.ts (hunk 2 above)
Comment (madepiet) [reply ts/parts/PieSeries.ts:1083]: I was just using it for a sample and forgot to change, very sorry for that.

**Comment by felipemfp** [comment]
@TorsteinHonsi when the release with this fix will be available?

**Pull request merged**